### PR TITLE
Add customer from dashboard

### LINKED ISSUE #1578: Allow creating customers from the dashboard
In the "Customers" view in the dashboard, we should have the ability to add new customers. It should work similarly to creating staff members, so after creating a new User instance, an email with password setup link should be sent.

### PR DESCRIPTION
**Pull request by koradon**
I want to merge this change because...

We want to have the possibility to add customers from the dashboard.
Fixes #1578 

screenshot from 2018-01-08 15-39-49
screenshot from 2018-01-08 15-39-33
screenshot from 2018-01-08 15-39-24
screenshot from 2018-01-08 15-39-14
screenshot from 2018-01-08 15-39-04

### Pull Request Checklist

(Please keep this section. It will make maintainer's life easier.)

1. [ ] Privileged views and APIs are guarded by proper permission checks.
1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] Database queries are optimized and the number of queries is constant.
1. [ ] The changes are tested.
1. [ ] The code is documented (docstrings, project documentation).
1. [ ] Python code quality checks pass: `pycodestyle`, `pydocstyle`, `pylint`.
1. [ ] JavaScript code quality checks pass: `eslint`.

### CODE REVIEW

**Comment by patrys** [comment]
It would be nice to know which use cases we're solving here.

**Comment by koradon** [comment]
Jup, added

**Comment by patrys** [comment]
It links to itself 😄

**Comment by koradon** [comment]
Lol done

File: saleor/dashboard/customer/forms.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+from ...userprofile.models import User
+
+
+class CustomerForm(forms.ModelForm):
+    def __init__(self, *args, **kwargs):
+        self.user = kwargs.pop('user', None)
+        super().__init__(*args, **kwargs)
+        if self.user == self.instance:
+            self.fields['is_active'].disabled = True
```
Comment (elwoodxblues) [inline saleor/dashboard/customer/forms.py]: We could add a comment here explaining why this field is disabled here.

File: saleor/dashboard/customer/views.py [hunk 2]
```diff
@@ -41,6 +43,28 @@ def customer_details(request, pk):
     return TemplateResponse(request, 'dashboard/customer/detail.html', ctx)
 
 
+@staff_member_required
+@permission_required('userprofile.edit_user')
+def customer_edit(request, pk=None):
+    if pk:
```
Comment (elwoodxblues) [inline saleor/dashboard/customer/views.py]: Don't we have separate views for editing and creating in other modules?

Comment (koradon) [reply saleor/dashboard/customer/views.py]: Right now in saleor, we have this mixed. For example, shipping_methods (and other former formsets) has one view for edit and create. This is sth I wanted to discuss with you.

File: saleor/dashboard/customer/forms.py (hunk 1 above)
Comment (koradon) [reply saleor/dashboard/customer/forms.py]: Ok

File: saleor/dashboard/customer/views.py (hunk 2 above)
Comment (elwoodxblues) [reply saleor/dashboard/customer/views.py]: So I think that it's much better to have separate views and I would consider those that do both creating and editing a technical debt.

Comment (koradon) [reply saleor/dashboard/customer/views.py]: I agree. Separate views for edit and create actions are more readable. Even if they are very similar.

**Comment by elwoodxblues** [comment]
@koradon Please fix conflicts.

**Comment by koradon** [comment]
@elwoodxblues rebase done

**Pull request merged**